tests/cpp/test_multidevice_sharding.cpp

```diff
@@ -684,14 +681,11 @@ TEST_F(MultiDeviceTest, ViewWithMerge) {
     tv->outer_split(0, d);
     tv->axis(0)->parallelize(ParallelType::DIDx);
   }
-  in->setAllocationDomain(in->getLoopDomain(), true);
+  // contiguity=false so the View won't be treated as a meta op and will
+  // trigger Pointwise, the purpose of the test.
+  in->setAllocationDomain(in->getLoopDomain(), false);
   out->setAllocationDomain(out->getLoopDomain(), true);
 
-  // So the View won't be treated as a meta op and will trigger Pointwise, the
-  // purpose of the test.
-  preseg_passes::OptimizationPassGuard<preseg_passes::MarkAliasesPreparePass>
-      optimization_guard(false);
-
```

Comment:
Why is disabling the aliasing pass not sufficient here?

Reply:
MAPP isn't always required to identify aliasing. I missed that when I initially wrote this test. MAPP inserts segmentsets and sets extra allocation orders. In this example, even without that, AA can tell alias between view in and out